jaeger-core/src/main/java/com/uber/jaeger/SpanContext.java

```diff
@@ -64,7 +63,7 @@ public SpanContext(long traceID, long spanID, long parentID, byte flags) {
 
   @Override
   public Iterable<Map.Entry<String, String>> baggageItems() {
-    return baggage.entrySet();
+    return new HashMap<String, String>(baggage).entrySet();
```

Comment:
ok, I am not too concerned with this method, it's rarely used anyway. The potential future improvement would be to use immutable map in the constructor.